### DISCUSSION TITLE
Calibration time bug

### DISCUSSION THREAD

**Discussion by iliyesku** [NightscoutFoundation/xDrip#2712] [Q&A]
![Screenshot_2023-03-01-20-58-47-781_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/20445960/222254556-9b209363-0485-4476-b67c-7a9463d528e9.jpg)
![IMG_20230301_210239](https://user-images.githubusercontent.com/20445960/222254602-f4f60a04-9fb8-4fac-96d8-d11de0b72b82.jpg)
After calibration at 20:23 calibration graph still passes through previous calibration point at 21:47 of the previous day.
Version: 344deb4-2022.11.16

**Comment by Navid200** [comment]
Is engineering mode enabled?

**Reply by Navid200** [reply]
Can you also enable data tables (https://navid200.github.io/xDrip/docs/Datatables.html) and show the calibration data table as well?

**Reply by iliyesku** [reply]
It is not in engineering mode.
![Screenshot_2023-03-03-06-55-33-995_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/20445960/222642840-772e8de9-e266-49da-b3b3-d6030e1b412e.jpg)

**Comment by PhilTU** [comment]
If it's a Libre sensor is it hitting the maximum intercept of 40 mg/dl and being rejected ? The 20:23 did, and the previous 21:47 was the last with an intercept below 40

**Reply by iliyesku** [reply]
The only I know it is the Libre 1 sensor

**Reply by PhilTU** [reply]
OK, so it probably is the limit. Calibration is rejected if it calculates an intercept over 40 mg/dl

**Reply by Navid200** [reply]
@PhilTU Thanks for identifying what was happening.
I wonder if we need to add a marker or note or log to xDrip to help user understands what xDrip is doing.
I mean there must be many other xDrip users who experience what @iliyesku has but not all of them post here to ask.

**Reply by PhilTU** [reply]
>  I wonder if we need to add a marker or note or log to xDrip to help user understands what xDrip is doing. 

Agreed, tagging it "Rejected" or coloured highlight perhaps.

I think it is especially problematic when a forced initial calibration results in a rejected calibration and hence no data on the screen.

**Reply by Navid200** [reply]
But, I don't know why this has not been done already.  There must be a reason.
@tzachi-dar What do you think?

**Reply by iliyesku** [reply]
Does calibration mean changing inside the sensor?

**Reply by Navid200** [reply]
This calibration is done in xDrip.

**Reply by iliyesku** [reply]
> This calibration is done in xDrip.

So, if it affects only xDrip internal mapping, why this restriction exists?

**Reply by Navid200** [reply]
A maximum intercept of 40mg/dL is a safety limit.  It cannot be removed.  It can result in a disaster:  Imagine someone having a low blood glucose but because of a faulty calibration, they get a report of very high glucose.  That should not be allowed.  That's why there is a max acceptable intercept.

**Reply by iliyesku** [reply]
> A maximum intercept of 40mg/dL is a safety limit.

It seems that this decision was not an easy one.

But what about the slope? It can still be adjusted for high values. ... I see that it can result a bigger deviation, and this brings me to another thought, why not use non-linear function?

**Reply by Navid200** [reply]
I personally wasn't involved when this decision was made.
I have asked @tzachi-dar to share his view here.  He will.  He knows a lot more than me about this.

**Reply by PhilTU** [reply]
> > A maximum intercept of 40mg/dL is a safety limit.
> 
> It seems that this decision was not an easy one.
> 
> But what about the slope? It can still be adjusted for high values. ... I see that it can result a bigger deviation, and this brings me to another thought, why not use non-linear function?

All things are possible. I think in practice the slopes were close to one but intercept more variable. As the sensor minimum output value is 40 an intercept of 40 with a slope of 1 means you never see a display value below 80 (a safe value), hence the danger as Navid explains if the measurement is 20 (certainly not safe).

**Comment by tzachi-dar** [comment]
There are a few things to say here:
As navid has explained, slope can not be bigger than 40. This is a security feature.
The thing that might be used to resolve the issue is to simply enter a calibration when the BG is low. I agree that there is probably a line that would have a low intercept and a higher slope.
If you ask me, it seems that the main issue here is that we have two points at 20:23 and 21:47 with the same raw value but with a big difference in BG values. One of the points is obviously wrong. xDrip gives a higher influence to the later point and I guess that this is the reason for the graph.
So, what one should probably be able to do in order to resolve the issue is either add a low BG calibration or remove the calibration from 21:47.

**Reply by iliyesku** [reply]
But 21:47 relates to the previous day, it can not be even deleted

**Reply by Navid200** [reply]
If xDrip rejects a calibration, it makes sense to mark it somehow on the calibration graph.
Otherwise, this question will keep coming up and it is a legitimate question.  A calibration shown on the calibration graph has no effect; why?
The answer is because xDrip rejected it.  But, why is it not clear on the graph that it has been rejected?
In my opinion, that is also a legitimate question.
